src/System.Console/src/System/IO/SyncTextReader.cs

```diff
@@ -31,8 +31,8 @@ protected override void Dispose(bool disposing)
             {
                 lock (_methodLock)
                 {
-                    // Explicitly pick up a potentially methodimpl'ed Dispose
-                    ((IDisposable)_in).Dispose();
+                    // This will always pick up TextReader.Dispose
```

Comment:
@nguerrera I don't see any effect this would have, actually. Neither `TextWriter` or `TextReader` explicitly impl IDisposable, and Dispose is resolved during compilation to pick up on those implementations of IDisposable. The original authors of the code may have made a mistake.